File: backends/go-gin/internal/usecase/gate/dto.go
package gate

// IssueDto は JWT 発行のユースケース入力です。
type IssueDto struct {
	AccessToken string
	MemberID    string
}

// VerifyDto は JWT 検証のユースケース入力です。
type VerifyDto struct {
	Identifier string
	Token      string
}

File: backends/ruby-rails/app/usecase/gate/dto.rb
# frozen_string_literal: true
#
# ゲートユースケースの DTO を定義するモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module UseCase
  module Gate
    # トークン発行時に渡す DTO です。
    IssueDto  = Struct.new(:access_token, :member_id, keyword_init: true)

    # トークン検証時に渡す DTO です。
    VerifyDto = Struct.new(:identifier, :token, keyword_init: true)
  end
end
